### PR TITLE
Criteo fix

### PR DESCRIPTION
**Pull request by znado**
Example run on tensorboard.dev [here](https://tensorboard.dev/experiment/sxRxQoTJQfKozuyvPd1BoQ/#scalars&_smoothingWeight=0)

- fixing criteo (re-fixing the loss fn shapes, correcting the default model sizings, correcting input pipeline)
- reducing criteo batch size by half so pytorch fits
- adding donate_argnums to jax submission base in ref algos
- allowing per-workload testing in reference algo tests
- logging rng seed
- cleaning up speech tokenizer init
- cleaning up dataset setup and updating criteo ds setup logic

### CODE REVIEW

**Comment by znado** [comment]
One thing that I need to investigate is whether or not we hit an OOM halfway through training while checkpointing.

File: algorithmic_efficiency/workloads/criteo1tb/workload.py [hunk 1]
```diff
@@ -14,17 +14,17 @@
...
-  embed_dim: int = 64
+  mlp_bottom_dims: Tuple[int, int] = (512, 256, 128)
+  mlp_top_dims: Tuple[int, int, int] = (1024, 1024, 512, 256, 1)
+  embed_dim: int = 128
 
   def has_reached_goal(self, eval_result: float) -> bool:
     return eval_result['validation/loss'] < self.target_value
 
   @property
-  def target_value(self) -> float:
```
Comment (runame) [inline algorithmic_efficiency/workloads/criteo1tb/workload.py:27]: Random question: is there a reason we don't want type hints here?

**Review by runame — APPROVED** [review]
LGTM!

File: algorithmic_efficiency/workloads/criteo1tb/input_pipeline.py [hunk 2]
```diff
@@ -6,92 +6,125 @@
...
 
-  if is_training:
-    ds = ds.repeat()
   ds = ds.interleave(
       tf.data.TextLineDataset,
-      cycle_length=32,
-      block_length=per_device_batch_size,
-      num_parallel_calls=32,
+      cycle_length=128,
+      block_length=global_batch_size // 8,
```
[the remark this answers is not in the document]
Comment (znado) [reply algorithmic_efficiency/workloads/criteo1tb/input_pipeline.py:117]: I think it's fine to keep as 8 for now.

File: algorithmic_efficiency/workloads/criteo1tb/workload.py (hunk 1 above)
Comment (znado) [reply algorithmic_efficiency/workloads/criteo1tb/workload.py:27]: nope, accidentally copied over these. done.

**Comment by znado** [comment]
Note that for pytorch I am currently experiencing some weird OOMs after evals, so that will have to be fixed in a follow-up PR, but I'm merging this for now to get the working jax changes in.
